File: js/src/04if.js
import {
    nextFloat
    , compareFloat
    , nextInt
    , compareInt
    , compareLine
    , initScanner
    , initChecker
    , inputsEOF
    , outputsEOF
} from './io.js'

(async () => {
    for (let taskNo = 1; taskNo <= 30; taskNo++) {
        const taskNoStr = String(taskNo).padStart(3, 0)
        for (let testNo = 1; testNo <= 100; testNo++) {
            const testNoStr = String(testNo).padStart(3, 0)
            const filePath = `../data/kt-gov2/04if/If${taskNoStr}/${testNoStr}/${testNoStr}`

            await initScanner(filePath + '.dat')
            await initChecker(filePath + '.ans')

            let ok = true
            switch (taskNo) {
                case 1: ok = if1(); break
                case 2: ok = if2(); break
                case 3: ok = if3(); break
                case 4: ok = if4(); break
                case 5: ok = if5(); break
                case 6: ok = if6(); break
                case 7: ok = if7(); break
                case 8: ok = if8(); break
                case 9: ok = if9(); break
                case 10: ok = if10(); break
                case 11: ok = if11(); break
                case 12: ok = if12(); break
                case 13: ok = if13(); break
                case 14: ok = if14(); break
                case 15: ok = if15(); break
                case 16: ok = if16(); break
                case 17: ok = if17(); break
                case 18: ok = if18(); break
                case 19: ok = if19(); break
                case 20: ok = if20(); break
                case 21: ok = if21(); break
                case 22: ok = if22(); break
                case 23: ok = if23(); break
                case 24: ok = if24(); break
                case 25: ok = if25(); break
                case 26: ok = if26(); break
                case 27: ok = if27(); break
                case 28: ok = if28(); break
                case 29: ok = if29(); break
                case 30: ok = if30(); break
            }

            if (!ok) {
                console.log("If" + taskNoStr + " #" + testNoStr + " has NOT been tested!")
                return
            }

            if (!inputsEOF()) {
                console.log("If" + taskNoStr + " #" + testNoStr + " input NOT EOF!")
                return
            }

            if (!outputsEOF()) {
                console.log("If" + taskNoStr + " #" + testNoStr + " output NOT EOF!")
                return
            }
        }
        console.log("If" + taskNoStr + " has been tested!")
    }
    console.log("If has been tested!")
})()

// { "no": 1, "dat": "", "ans": "" }
const if1 = () => {
    let number = nextInt()
    if (number > 0) {
        number -= 8
    }
    return compareInt(number)
}

// { "no": 2, "dat": "", "ans": "" }
const if2 = () => {
    let number = nextInt()
    if (number > 0) {
        number -= 8
    } else {
        number += 6
    }
    return compareInt(number)
}

// { "no": 3, "dat": "", "ans": "" }
const if3 = () => {
    let number = nextInt()
    if (number > 0) {
        number -= 8
    } else if (number < 0) {
        number += 6
    } else {
        number = 10
    }
    return compareInt(number)
}

// { "no": 4, "dat": "", "ans": "" }
const if4 = () => {
    let a = nextInt()
    let b = nextInt()
    let c = nextInt()
    let positivesCount = 0
    if (a > 0) {
        positivesCount++
    }
    if (b > 0) {
        positivesCount++
    }
    if (c > 0) {
        positivesCount++
    }
    return compareInt(positivesCount)
}

// { "no": 5, "dat": "", "ans": "" }
const if5 = () => {
    let a = nextInt()
    let b = nextInt()
    let c = nextInt()
    let positivesCount = 0, negativesCount = 0
    if (a > 0) {
        positivesCount++
    } else if (a < 0) {
        negativesCount++
    }
    if (b > 0) {
        positivesCount++
    } else if (b < 0) {
        negativesCount++
    }
    if (c > 0) {
        positivesCount++
    } else if (c < 0) {
        negativesCount++
    }
    return compareInt(positivesCount, negativesCount)
}

// { "no": 6, "dat": "2", "ans": "2" }
const if6 = () => {
    let a = nextFloat()
    let b = nextFloat()
    let larger = a
    if (b > a) {
        larger = b
    }
    return compareFloat(2, larger)
}

// { "no": 7, "dat": "2", "ans": "" }
const if7 = () => {
    let a = nextFloat()
    let b = nextFloat()
    let smallerIndex = 1
    if (b < a) {
        smallerIndex = 2
    }
    return compareInt(smallerIndex)
}

// { "no": 8, "dat": "2", "ans": "2" }
const if8 = () => {
    let a = nextFloat()
    let b = nextFloat()
    let larger = a
    if (b > a) {
        larger = b
    }
    let smaller = a + b - larger
    return compareFloat(2, larger, smaller)
}

// { "no": 9, "dat": "2", "ans": "2" }
const if9 = () => {
    let a = nextFloat()
    let b = nextFloat()
    if (a > b) {
        a += b
        b = a - b
        a -= b
    }
    return compareFloat(2, a, b)
}

// { "no": 10, "dat": "", "ans": "" }
const if10 = () => {
    let a = nextInt()
    let b = nextInt()
    if (a != b) {
        b = a += b
    } else {
        a = b = 0
    }
    return compareInt(a, b)
}

// { "no": 11, "dat": "", "ans": "" }
const if11 = () => {
    let a = nextInt()
    let b = nextInt()
    if (a != b) {
        let larger = a
        if (b > a) {
            larger = b
        }
        a = b = larger
    } else {
        a = b = 0
    }
    return compareInt(a, b)
}

// { "no": 12, "dat": "2", "ans": "2" }
const if12 = () => {
    let a = nextFloat()
    let b = nextFloat()
    let c = nextFloat()
    let minimal
    if (a < b && a < c) {
        minimal = a
    } else if (b < c) {
        minimal = b
    } else {
        minimal = c
    }
    return compareFloat(2, minimal)
}

// { "no": 13, "dat": "2", "ans": "2" }
const if13 = () => {
    let a = nextFloat()
    let b = nextFloat()
    let c = nextFloat()
    let minimal, maximal
    if (a < b && a < c) {
        minimal = a
    } else if (b < c) {
        minimal = b
    } else {
        minimal = c
    }
    if (a > b && a > c) {
        maximal = a
    } else if (b > c) {
        maximal = b
    } else {
        maximal = c
    }
    let between = a + b + c - minimal - maximal
    return compareFloat(2, between)
}

// { "no": 14, "dat": "2", "ans": "2" }
const if14 = () => {
    let a = nextFloat()
    let b = nextFloat()
    let c = nextFloat()
    let minimal, maximal
    if (a < b && a < c) {
        minimal = a
    } else if (b < c) {
        minimal = b
    } else {
        minimal = c
    }
    if (a > b && a > c) {
        maximal = a
    } else if (b > c) {
        maximal = b
    } else {
        maximal = c
    }
    return compareFloat(2, minimal, maximal)
}

// { "no": 15, "dat": "2", "ans": "2" }
const if15 = () => {
    let a = nextFloat()
    let b = nextFloat()
    let c = nextFloat()
    let sum = a + b + c
    if (a < b && a < c) {
        sum -= a
    } else if (b < c) {
        sum -= b
    } else {
        sum -= c
    }
    return compareFloat(2, sum)
}

// { "no": 16, "dat": "2", "ans": "2" }
const if16 = () => {
    let a = nextFloat()
    let b = nextFloat()
    let c = nextFloat()
    let factor
    if (a < b && b < c) {
        factor = 2
    } else {
        factor = -1
    }
    a *= factor
    b *= factor
    c *= factor
    return compareFloat(2, a, b, c)
}

// { "no": 17, "dat": "2", "ans": "2" }
const if17 = () => {
    let a = nextFloat()
    let b = nextFloat()
    let c = nextFloat()
    let factor
    if (a < b && b < c || a > b && b > c) {
        factor = 2
    } else {
        factor = -1
    }
    a *= factor
    b *= factor
    c *= factor
    return compareFloat(2, a, b, c)
}

// { "no": 18, "dat": "", "ans": "" }
const if18 = () => {
    let a = nextInt()
    let b = nextInt()
    let c = nextInt()
    let differIndex
    if (a == b) {
        differIndex = 3
    } else if (a == c) {
        differIndex = 2
    } else {
        differIndex = 1
    }
    return compareInt(differIndex)
}

// { "no": 19, "dat": "", "ans": "" }
const if19 = () => {
    let a = nextInt()
    let b = nextInt()
    let c = nextInt()
    let d = nextInt()
    let differIndex
    if (a == b && b == c) {
        differIndex = 4
    } else if (d != c) {
        differIndex = 3
    } else if (d != b) {
        differIndex = 2
    } else {
        differIndex = 1
    }
    return compareInt(differIndex)
}

// { "no": 20, "dat": "2", "ans": "2" }
const if20 = () => {
    let a = nextFloat()
    let b = nextFloat()
    let c = nextFloat()
    let distance, point
    if (Math.abs(b - a) < Math.abs(c - a)) {
        point = b
    } else {
        point = c
    }
    distance = Math.abs(point - a)
    return compareFloat(2, point, distance)
}

// { "no": 21, "dat": "", "ans": "" }
const if21 = () => {
    let x = nextInt()
    let y = nextInt()
    let result
    if (x == 0 && y == 0) {
        result = 0
    } else if (y == 0) {
        result = 1
    } else if (x == 0) {
        result = 2
    } else {
        result = 3
    }
    return compareInt(result)
}

// { "no": 22, "dat": "2", "ans": "" }
const if22 = () => {
    let x = nextFloat()
    let y = nextFloat()
    let quarter
    if (x > 0) {
        if (y > 0) {
            quarter = 1
        } else {
            quarter = 4
        }
    } else if (y > 0) {
        quarter = 2
    } else {
        quarter = 3
    }
    return compareInt(quarter)
}

// { "no": 23, "dat": "", "ans": "" }
const if23 = () => {
    let x1 = nextInt()
    let y1 = nextInt()
    let x2 = nextInt()
    let y2 = nextInt()
    let x3 = nextInt()
    let y3 = nextInt()
    let x4, y4
    if (x1 == x2) {
        x4 = x3
    } else if (x1 == x3) {
        x4 = x2
    } else {
        x4 = x1
    }
    if (y1 == y2) {
        y4 = y3
    } else if (y1 == y3) {
        y4 = y2
    } else {
        y4 = y1
    }
    return compareInt(x4, y4)
}

// { "no": 24, "dat": "2", "ans": "2" }
const if24 = () => {
    let x = nextFloat()
    let fx
    if (x > 0) {
        fx = 2 * Math.sin(x)
    } else {
        fx = 6 - x
    }
    return compareFloat(2, fx)
}

// { "no": 25, "dat": "", "ans": "" }
const if25 = () => {
    let x = nextInt()
    let fx
    if (x < -2 || x > 2) {
        fx = 2 * x
    } else {
        fx = -3 * x
    }
    return compareInt(fx)
}

// { "no": 26, "dat": "2", "ans": "2" }
const if26 = () => {
    let x = nextFloat()
    let fx
    if (x <= 0) {
        fx = -x
    } else if (x >= 2) {
        fx = 4
    } else {
        fx = x * x
    }
    return compareFloat(2, fx)
}

// { "no": 27, "dat": "2", "ans": "" }
const if27 = () => {
    let x = nextFloat()
    let fx
    if (x < 0) {
        fx = 0
    } else if (parseInt(x) % 2 == 0) {
        fx = 1
    } else {
        fx = -1
    }
    return compareInt(fx)
}

// { "no": 28, "dat": "", "ans": "" }
const if28 = () => {
    let year = nextInt()
    let daysAmount
    if (year % 400 == 0) {
        daysAmount = 366
    } else if (year % 100 == 0) {
        daysAmount = 365
    } else if (year % 4 == 0) {
        daysAmount = 366
    } else {
        daysAmount = 365
    }
    return compareInt(daysAmount)
}

// { "no": 29, "dat": "", "ans": "" }
const if29 = () => {
    let number = nextInt()
    let description = ""
    if (number == 0) {
        description += "zero"
    } else {
        if (number > 0) {
            description += "positive"
        } else {
            description += "negative"
        }
        if (number % 2 == 0) {
            description += " even"
        } else {
            description += " odd"
        }
    }
    description += " number"
    return compareLine(description)
}

// { "no": 30, "dat": "", "ans": "" }
const if30 = () => {
    let number = nextInt()
    let description = ""
    if (number > 0 && number < 1000) {
        if (number < 10) {
            description += "one"
        } else if (number < 100) {
            description += "two"
        } else {
            description += "three"
        }
        description += "-digit "
        if (number % 2 == 0) {
            description += "even"
        } else {
            description += "odd"
        }
        description += " number"
    }
    return compareLine(description)
}

File: java/04if.java
import java.util.Scanner;

class IfTask {

	private boolean if1() {
		Scanner scanner = new Scanner(System.in);
		System.out.print("number = ");
		int number = scanner.nextInt();
		if (number > 0) {
			number -= 8;
		}
		System.out.println("number = " + number);
		return false;
	}

	private boolean if2() {
		Scanner scanner = new Scanner(System.in);
		System.out.print("number = ");
		int number = scanner.nextInt();
		if (number > 0) {
			number -= 8;
		} else {
			number += 6;
		}
		System.out.println("number = " + number);
		return false;
	}

	private boolean if3() {
		Scanner scanner = new Scanner(System.in);
		System.out.print("number = ");
		int number = scanner.nextInt();
		if (number > 0) {
			number -= 8;
		} else if (number < 0) {
			number -= 6;
		} else {
			number = 10;
		}
		System.out.println("number = " + number);
		return false;
	}

	private boolean if4() {
		Scanner scanner = new Scanner(System.in);
		System.out.print("a = ");
		int a = scanner.nextInt();
		System.out.print("b = ");
		int b = scanner.nextInt();
		System.out.print("c = ");
		int c = scanner.nextInt();
		int positiveCount = 0;
		if (a > 0) {
			positiveCount++;
		}
		if (b > 0) {
			positiveCount++;
		}
		if (c > 0) {
			positiveCount++;
		}
		System.out.println("positiveCount = " + positiveCount);
		return false;
	}

	private boolean if5() {
		Scanner scanner = new Scanner(System.in);
		System.out.print("a = ");
		int a = scanner.nextInt();
		System.out.print("b = ");
		int b = scanner.nextInt();
		System.out.print("c = ");
		int c = scanner.nextInt();
		int positiveCount = 0;
		int negativeCount = 0;
		if (a > 0) {
			positiveCount++;
		} else if (a < 0) {
			negativeCount++;
		}
		if (b > 0) {
			positiveCount++;
		} else if (b < 0) {
			negativeCount++;
		}
		if (c > 0) {
			positiveCount++;
		} else if (c < 0) {
			negativeCount++;
		}
		System.out.println("positiveCount = " + positiveCount);
		System.out.println("negativeCount = " + negativeCount);
		return false;
	}

	private boolean if6() {
		Scanner scanner = new Scanner(System.in);
		System.out.print("number1 = ");
		double number1 = scanner.nextDouble();
		System.out.print("number2 = ");
		double number2 = scanner.nextDouble();
		if (number1 > number2) {
			System.out.println(number1);
		} else {
			System.out.println(number2);
		}
		return false;
	}

	private boolean if7() {
        double a = getDouble();
        double b = getDouble();
        if (a < b) {
            put(1);
        } else {
            put(2);
        }
        return false;
    }
	
	private boolean if8() {
        double A = getDouble();
        double B = getDouble();
        if (A > B) {
            put(A);
            put(B);
        } else {
            put(B);
            put(A);
        }
        return false;
    }

	private boolean if9() {
        double A = getDouble();
        double B = getDouble();
        if (A > B) {
            double temp = A;
            A = B;
            B = temp;
        }
        put(A);
        put(B);
        return false;
    }

	private boolean if10() {
        int A = getInt();
        int B = getInt();
        if (A != B) {
            A += B;
            B = A;
        } else {
            A = B = 0;
        }
        put(A);
        put(B);
        return false;
    }

	private boolean if11() {
		int A = getInt();
		int B = getInt();
		if (A != B) {
			if (A > B) {
				B = A;
			} else {
				A = B;
			}
		} else {
			A = B = 0;
		}
		put(A);
		put(B);
		return false;
	}

	private boolean if12() {
        double a = getDouble();
        double b = getDouble();
        double c = getDouble();
        double xurd = 0;
        if ( (a < b) && (a < c) ) {
            xurd = a;
        } else if (b < c) {
            xurd = b;
        } else {
            xurd = c;
        }
        put(xurd);
        return false;
    }

	private boolean if13() {
        double a = getDouble();
        double b = getDouble();
        double c = getDouble();
        double kalon, xurd;
        if ( (a > b) && (a > c) ) {
            kalon = a;
        } else if (b > c) {
            kalon = b;
        } else {
            kalon = c;
        }
        if ( (a < b) && (a < c) ) {
            xurd = a;
        } else if (b < c) {
            xurd = b;
        } else {
            xurd = c;
        }
        double middle = a+b+c - kalon - xurd;
        put(middle);
        return false;
    }

	private boolean if14() {
        double a = getDouble();
        double b = getDouble();
        double c = getDouble();
        double kalon, xurd;
        if ( (a > b) && (a > c) ) {
            kalon = a;
        } else if (b > c) {
            kalon = b;
        } else {
            kalon = c;
        }
        if ( (a < b) && (a < c) ) {
            xurd = a;
        } else if (b < c) {
            xurd = b;
        } else {
            xurd = c;
        }
        put(xurd);
        put(kalon);
        return false;
    }

	private boolean if15() {
        double a = getDouble();
        double b = getDouble();
        double c = getDouble();
        double xurd = 0;
        if ( (a < b) && (a < c) ) {
            xurd = a;
        } else if (b < c) {
            xurd = b;
        } else {
            xurd = c;
        }
        double sum = a + b + c - xurd;
        put(sum);
        return false;
    }

	private boolean if16() {
        double A = getDouble();
        double B = getDouble();
        double C = getDouble();
        if ( (A < B) && (B < C) ) {
            A *= 2;
            B *= 2;
            C *= 2;
        } else {
            A *= -1;
            B *= -1;
            C *= -1;
        }
        put(A);
        put(B);
        put(C);
        return false;
    }

	private boolean if17() {
        double A = getDouble();
        double B = getDouble();
        double C = getDouble();
        if ( (A < B) && (B < C) || (A > B) && (B > C) ) {
            A *= 2;
            B *= 2;
            C *= 2;
        } else {
            A *= -1;
            B *= -1;
            C *= -1;
        }
        put(A);
        put(B);
        put(C);
        return false;
    }

	private boolean if18() {
        int A = getInt();
        int B = getInt();
        int C = getInt();
        if (A == B) {
            put(3);
        } else if (A == C) {
            put(2);
        } else {
            put(1);
        }
        return false;
    }

	private boolean if19() {
        int A = getInt();
        int B = getInt();
        int C = getInt();
        int D = getInt();
        if (A == B) {
            if (C != A) {
                put(3);
            } else {
                put(4);
            }
        } else {
            if (A == C) {
                put(2);
            } else {
                put(1);
            }
        }
        return false;
    }

	private boolean if20() {
        double A = getDouble();
        double B = getDouble();
        double C = getDouble();
        double AB = Math.abs(B - A);
        double AC = Math.abs(C - A);
        if (AB < AC) {
            put(B);
            put(AB);
        } else {
            put(C);
            put(AC);
        }
        return false;
    }

	private boolean if21() {
        int x = getInt();
        int y = getInt();
        if ( (x == 0) && (y == 0) ) {
            put(0);
        } else if (y == 0) {
            put(1);
        } else if (x == 0) {
            put(2);
        } else {
            put(3);
        }
        return false;
    }

	private boolean if22() {
        double x = getDouble();
        double y = getDouble();
        int quarterNo = 0;
        if (x > 0) {
            if (y > 0) {
                quarterNo = 1;
            } else {
                quarterNo = 4;
            }
        } else {
            if (y > 0) {
                quarterNo = 2;
            } else {
                quarterNo = 3;
            }
        }
        put(quarterNo);
        return false;
    }

	private boolean if23() {
        int x1 = getInt();
        int y1 = getInt();
        int x2 = getInt();
        int y2 = getInt();
        int x3 = getInt();
        int y3 = getInt();
        int x4 = 0, y4 = 0;
        if (x1 == x2) {
            x4 = x3;
        } else if (x1 == x3) {
            x4 = x2;
        } else {
            x4 = x1;
        }
        if (y1 == y2) {
            y4 = y3;
        } else if (y1 == y3) {
            y4 = y2;
        } else {
            y4 = y1;
        }
        put(x4);
        put(y4);
        return false;
    }

	private boolean if24() {
        double x = getDouble();
        double y = 0;
        if (x > 0) {
            y = 2 * Math.sin(x);
        } else {
            y = 6 - x;
        }
        put(y);
        return false;
    }

	private boolean if25() {
        int x = getInt();
        int y = 0;
        if ( (x < -2) || (x > 2) ) {
            y = 2*x;
        } else {
            y = -3*x;
        }
        put(y);
        return false;
    }

	private boolean if26() {
        double x = getDouble();
        double y = 0;
        if (x <= 0) {
            y = -x;
        } else if (x >= 2) {
            y = 4;
        } else {
            y = x*x;
        }
        put(y);
        return false;
    }

	private boolean if27() {
        double x = getDouble();
        int y = 0;
        if (x < 0) {
            y = 0;
        } else if ((int)x % 2 == 0) {
            y = 1;
        } else {
            y = -1;
        }
        put(y);
        return false;
    }

	private boolean if28() {
        int year = getInt();
        int days = 0;
        if (year % 400 == 0) {
            days = 366;
        } else if (year % 100 == 0) {
            days = 365;
        } else if (year % 4 == 0) {
            days = 366;
        } else {
            days = 365;
        }
        put(days);
        return false;
    }

	private boolean if29() {
        int number = getInt();
        String descr = "";
        if (number == 0) {
            descr += "нулевое ";
        } else {
            if (number > 0) {
                descr += "положительное ";
            } else {
                descr += "отрицательное ";
            }
            if (number % 2 != 0) {
                descr += "не";
            }
            descr += "четное ";
        }
        descr += "число";
        put(descr);
        return false;
    }

	private boolean if30() {
        int number = getInt();
        String descr = "";
        if (number % 2 != 0) {
            descr += "не";
        }
        descr += "четное ";
        if (number <= 9) {
            descr += "одно";
        } else if (number <= 99) {
            descr += "дву";
        } else {
            descr += "трех";
        }
        descr += "значное число";
        put(descr);
        return false;
    }
}